enhancements/agent-installer/simplified-registry-less-cluster-operations.md

```diff
@@ -0,0 +1,522 @@
+necessary elements to support the cluster operations in a disconnected
+environment.
+This ISO will be prepared and built using the [OpenShift Appliance Builder](https://github.com/openshift/appliance),
+a command line utility for building a disk image to orchestrate an OpenShift
+installation. The Appliance builder will be enhanced with a new 
+`build live-iso` command capable to generate an ISO artifact. 
+The extend ISO builder will be integrated within the official Red Hat build and
+release pipeline, so that a new extended RHCOS ISO could be published following
+the same OpenShift release cadence, and made it available via the 
+[Red Hat Customer Portal](https://access.redhat.com).
+
+### Installation
+
+#### Bootstrap phase
+
+Based on the same [Agent-based Installer](https://github.com/openshift/enhancements/blob/master/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md)(ABI)
+approach, the installation will be kicked off into an ephemeral environment
+fully allocated in memory, as soon as each node is booted using the extended
+ISO (bootstrap phase).
+In addition to the usual ABI services configured to orchestrate the
+installation, a local registry service will be created on each node, and the
+content of the release payload from the mounted ISO will be used as its
+backend source.
+During this early phase, the main responsibility of the local registry will
+be to serve any image pull request originating from the local services running
+on the same node.
+
+#### Installation phase (after the first reboot)
+
+During the installation of each control plane node, the contents of the OCP
+release payload will be copied from the ISO to the node's disk (in the format
+of the registry's 'filesystem' backend) after the OS disk writing step - so
+that it remains available not only after the first reboot, but also once the
+installation is complete. Also the registry container image itself will be
+copied in the node additional image store for podman.
+[Assisted Service](https://github.com/openshift/assisted-service/)
+and [Assisted Installer](https://github.com/openshift/assisted-installer)
+will be enhanced accordingly to take care of this step, and to ensure it
+will be properly monitored during the installation process.
+
+A new systemd service will be injected via MCO to run the registry container
+for providing the release content stored on disk after the reboot (since the
+previous ephemeral environment will not be available anymore).
+
+To allow pulling images from the cluster, an ImageDigestMirrorSet
+(IDMS) policy will be added to the cluster manifests - using the `api-int` 
+DNS entry - so that the set of registries running within
+the control plane nodes will transparently serve the release payload images
+locally mirrored.
+
+Finally, a new `InternalReleaseImage` custom resource, managed by MCO, will
+also be added as a cluster manifest to keep track and manage the currently
+stored internal release image.
+
```

Comment:
Add details of the OLM aspect of the installation here?

Reply:
Do you maybe mean reporting the list of what was exactly included in the current release bundle? If so I will sooner update the proposal with more details on this topic (see also this [comment](https://github.com/openshift/enhancements/pull/1821/files#r2342887223))

Reply:
cc @danmanor for additional comments